### DISCUSSION TITLE
How to include maap.py package in custom environment for DPS jobs?

### DISCUSSION THREAD

**Discussion by arojas314** [MAAP-Project/Community#802] [Q&A] [answered]
Hi all,

How would I include the maap.py pacakge in custom conda environment in DPS jobs? Registering algorithm through the DPS page:
Build command script:
```bash
#!/bin/bash

# source activate base
basedir=$( cd "$(dirname "$0")" ; pwd -P)
conda env list
echo installing environment...
conda env create -f ${basedir}/environment.yml
```
**environment.yml** was created using `conda env export > environment.yml`

I followed the video walkthrough (MAAP UWG Workshop-20230615_113428-Meeting Recording.mp4) but it was not clear.

**Comment by arojas314** [comment]
Noticed that the environment.yml script lists maap-py package under pip, but it is not available through pip.
It was suggested to clone maap-py repository and install using:
```python
# source activate base
basedir=$( cd "$(dirname "$0")" ; pwd -P)
echo installing environment...
mamba env create -f ${basedir}/environment.yml

# Install the maap.py environment
source activate custom-env
git clone --single-branch --branch v3.0.1 https://github.com/MAAP-Project/maap-py.git ${basedir}
cd ${basedir}/maap-py
pip install -e .
```
But i received the following error when registering algorithm:
```bash
fatal: destination path '/app/biomass-gedi-conus' already exists and is not an empty directory.
biomass-gedi-conus/build-env.sh: line 11: cd: /app/biomass-gedi-conus/maap-py: No such file or directory
Obtaining file:///app
ERROR: file:///app does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
Removing intermediate container 938163f78943
The command '/bin/sh -c bash ${BUILD_CMD}' returned a non-zero code: 1
```

**Reply by sujen1412** [reply, marked as answer]
I think there is an issue in your clone command that specifies ${basedir}. Its trying to overwrite it. Try the following

```
# source activate base
basedir=$( cd "$(dirname "$0")" ; pwd -P)
echo installing environment...
mamba env create -f ${basedir}/environment.yml

# Install the maap.py environment
source activate custom-env
pushd $basedir
git clone --single-branch --branch v3.0.1 https://github.com/MAAP-Project/maap-py.git
cd maap-py
pip install -e .

```

**Reply by sujen1412** [reply]
You can also pip install using git urls, many users have had success doing that. 
Have a look at https://github.com/lauraduncanson/icesat2_boreal/blob/master/dps/build_command_main.sh#L16. 

For now, try with the above git clone method to get the v3.0.1 version.

**Reply by arojas314** [reply]
Your first solution works, thanks for the help!
